### PR TITLE
sql: update doc for split partition table

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

Update doc for the `split table` statement.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/2810
- Other reference link(s):

### CODE REVIEW

File: reference/sql/statements/split-region.md [hunk 1]
```diff
@@ -32,6 +32,19 @@ SPLIT TABLE table_name [INDEX index_name] BY (value_list) [, (value_list)] ...
...
+```sql
++--------------------+----------------------+
+| TOTAL_SPLIT_REGION | SCATTER_FINISH_RATIO |
++--------------------+----------------------+
+| 4                  | 1.0                  |
++--------------------+----------------------+
+```
+
+* `TOTAL_SPLIT_REGION`: the number of newly split Regions.
+* `SCATTER_FINISH_RATIO`: the scattering rate of the newly split Regions. `1.0` means that all Regions are scattered. `0.5` means that only half of the Regions are scattered and the rest are being scattered.
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:46]:
```suggestion
* `SCATTER_FINISH_RATIO`: the completion rate of scattering for newly split Regions. `1.0` means that all Regions are scattered. `0.5` means that only half of the Regions are scattered and the rest are being scattered.
```

File: reference/sql/statements/split-region.md [hunk 2]
```diff
@@ -173,6 +186,155 @@ region3  [("b", "2019-04-17 14:26:19") , ("c", "")                   )
 region4  [("c", "")                    , maxIndexValue               )
 ```
 
+### Split Regions for partitioned tables
+
+Splitting Regions for partitioned tables is the same as splitting Regions for normal tables. The only difference is that the split operation is performed for every partition.
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:191]:
```suggestion
Splitting Regions for partitioned tables is the same as splitting Regions for ordinary tables. The only difference is that the same split operation is performed for every partition.
```

File: reference/sql/statements/split-region.md [hunk 3]
```diff
@@ -173,6 +186,155 @@ region3  [("b", "2019-04-17 14:26:19") , ("c", "")                   )
...
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    SPLIT [PARTITION] TABLE table_name [PARTITION (partition_name_list...)] [INDEX index_name] BY (value_list) [, (value_list)] ...
+    ```
+
+#### Examples
+
+Create a partitioned table `t`:
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:211]: More commits are made to https://github.com/pingcap/docs-cn/pull/2810. Please refer to the latest version and update this PR accordingly.

**Comment by TomShawn** [comment]
@lilin90 All comments are addressed, PTAL again, thanks!

File: reference/sql/statements/split-region.md [hunk 4]
```diff
@@ -173,6 +186,153 @@ region3  [("b", "2019-04-17 14:26:19") , ("c", "")                   )
...
+
+1. Create a partitioned table `t`.
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    create table t (a int,b int,index idx(a)) partition by hash(a) partitions 2;
+    ```
+
+    After creating the table `t`, a Region is split for each partition. Use the `SHOW TABLE REGIONS` syntax to see the Region of this table:
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:219]:
```suggestion
    After creating the table `t`, a Region is split for each partition. Use the `SHOW TABLE REGIONS` syntax to view the Regions of this table:
```

File: reference/sql/statements/split-region.md [hunk 5]
```diff
@@ -173,6 +186,153 @@ region3  [("b", "2019-04-17 14:26:19") , ("c", "")                   )
...
+
+2. Use the `SPLIT` syntax to split a Region for each partition. In the following example, four Regions are split in the range of `[0,10000]`.
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    split partition table t between (0) and (10000) regions 4;
+    ```
+
+3. Use the `SHOW TABLE REGIONS` syntax to see the Region of this table again. You can see that this table now has ten Regions, each partition with five Regions, four of which are the row data and one is the index data.
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:244]:
```suggestion
3. Use the `SHOW TABLE REGIONS` syntax to view the Regions of this table again. You can see that this table now has ten Regions, each partition with five Regions, four of which are the row data and one is the index data.
```

File: reference/sql/statements/split-region.md [hunk 6]
```diff
@@ -173,6 +186,153 @@ region3  [("b", "2019-04-17 14:26:19") , ("c", "")                   )
...
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    split partition table t index idx between (1000) and (10000) regions 2;
+    ```
+
+#### Split Region for a single partition
+
+You can specify the partition to be split. See the following steps for example:
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:279]:
```suggestion
You can specify the partition to be split. See the following usage example:
```

File: reference/sql/statements/split-region.md [hunk 7]
```diff
@@ -173,6 +186,153 @@ region3  [("b", "2019-04-17 14:26:19") , ("c", "")                   )
...
+
+3. Split two Regions in the `[10000,20000]` range of the `p2` partition:
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    split partition table t partition (p2) between (10000) and (20000) regions 2;
+    ```
+
+4. Use the `SHOW TABLE REGIONS` syntax to see the Region of this table:
```
Comment (lilin90) [inline reference/sql/statements/split-region.md:308]: From the results you can see there are multiple Regions.
```suggestion
4. Use the `SHOW TABLE REGIONS` syntax to view the Regions of this table:
```

**Pull request merged**